### PR TITLE
Building an API pt. 8 - lesson 3: New lesson - D.R.Ying Up Our Routes

### PR DESCRIPTION
**Pull request by kelsey-steven-ada**
[Pt. 8 Asana ticket](https://app.asana.com/0/1205655776549324/1206697443678574/f)
[Pt. 8 Lesson 3 ticket](https://app.asana.com/0/1205655776549324/1207017851991915/f)

Adds brand new lesson to Pt. 8 that covers refactoring similar or repeated model creation and model filtering code to helper functions.

### CODE REVIEW

File: api-8-relationships-in-sqlalchemy/drying-our-routes.md [hunk 1]
```diff
@@ -0,0 +1,613 @@
...
+- `POST` to `/authors`
+- `GET` to `/authors`
+- `POST` to `/authors/<author_id>/books`
+- `GET` to `authors/<author_id>/books`
+
+The `Book` model and table should have the following columns:
+- `id`
+- `title`
+- `description`
+- `author_id` (model only)
```
Comment (kelsey-steven-ada) [inline api-8-relationships-in-sqlalchemy/drying-our-routes.md:43]:
```suggestion
- `author_id` (foreign key)
```

File: api-8-relationships-in-sqlalchemy/drying-our-routes.md [hunk 2]
```diff
@@ -0,0 +1,613 @@
...
+
+What happens when a filter is applied vs when a filter is ignored?
+
+##### !end-hint
+##### !explanation
+
+a| False: we do not send back anything in the response that lest the user know which filters were applied
+b| True
+c| True
+d| True
```
Comment (mikellewade) [inline api-8-relationships-in-sqlalchemy/drying-our-routes.md:561]: For some reason Learn didn't apply the formatting correctly, I tried this and looks like it worked. Feel free to tweak it though! 

Screenshot 2024-10-04 at 11 38 20 AM

Screenshot 2024-10-04 at 11 40 13 AM

```suggestion
**a**) False: We do not send back anything in the response that lets the user know which filters were applied.\
**b**) True\
**c**) True\
**d**) True
```

File: api-8-relationships-in-sqlalchemy/drying-our-routes.md [hunk 3]
```diff
@@ -0,0 +1,613 @@
...
+
+Where does the function that constructs a `Book` from a dictionary live?
+
+##### !end-hint
+##### !explanation
+
+a| False: we could likely combine some statements to make our files even shorter, but shortness is not the goal; clear, easy to maintain code is the goal.
+b| True
+c| True
+d| False: If attributes on the `Book` model change, then we would need to make changes to the `to_dict` method of the `Book` class.
```
Comment (mikellewade) [inline api-8-relationships-in-sqlalchemy/drying-our-routes.md:608]: Same issue with these from [here](https://github.com/Ada-Developers-Academy/core-unit-2/pull/172/files#r1787905589).

Comment (kelsey-steven-ada) [reply api-8-relationships-in-sqlalchemy/drying-our-routes.md:608]: Ooo, thank you! This came up in a later PR & I forgot to come back here to see if it needed to be updated. I'll switch both the question explanations over to numbered lists.

**Review by mikellewade — APPROVED** [review]
LGTM ✅

**Pull request merged**